.github/workflows/ci.yml

```
@@ -0,0 +1,86 @@
+name: CI
+on: [pull_request, push]
+
+jobs:
+
+  linux:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - run: pip install meson ninja
+    - run: |
+        sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
+        sudo wget https://apt.llvm.org/llvm.sh
+        sudo chmod +x llvm.sh
+        sudo ./llvm.sh 15 all
+        sudo apt install libgtest-dev libbenchmark-dev ninja-build pkg-config cmake
+    - run: meson setup builddir/
+      env:
+        CC: gcc
+    - run: meson test -C builddir/
+    - uses: actions/upload-artifact@v1
+      if: failure()
+      with:
+        name: Linux_Meson_Testlog
+        path: builddir/meson-logs/testlog.txt
+
+  macos:
+    # env:
+
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - run: brew install gcc llvm@15 googletest google-benchmark ninja pkg-config cmake
+    - run: pip install meson ninja
+    - run: |
+        export SDKROOT=$(xcrun --show-sdk-path)
+        export PATH=$HOME/.local/bin:$PATH
+        export LD_LIBRARY_PATH=$HOME/.local/lib/x86_64-unknown-linux-gnu/:$HOME/.local/lib:$LD_LIBRARY_PATH
+        export PKG_CONFIG_PATH=$HOME/.local/lib/pkgconfig:$PKG_CONFIG_PATH
+        export C_INCLUDE_PATH=$HOME/.local/include:$C_INCLUDE_PATH
+        export CPLUS_INCLUDE_PATH=$HOME/.local/include/c++/v1:$HOME/.local/include:$CPLUS_INCLUDE_PATH
+    - run: meson setup builddir
+      env:
+        CC: gcc-12
+        LDFLAGS: "-L/usr/local/opt/llvm/lib"
+        CPPFLAGS: "-I/usr/local/opt/llvm/include"
+        # CC_LD: lld
+        # CXX_LD: lld
+        CXXFLAGS: "-stdlib: libc++" 
```

```suggestion
CXXFLAGS: "-stdlib=libstdc++"
```
`libstdc++` is the GNU C++ standard library, while `libc++` is the LLVM standard library.
The README suggests installing a new LLVM toolchain and using the standard library that comes with it.

I'm guessing I started using more new features (ranges, in particular) since then, and thus it stopped working.
The GNU standard library is what I'd been using locally on all my (Linux) computers, so it should work if you go through the trouble of installing it + gcc.

I'll try setting up my mac mini or maybe log back into deepsea to see if I can get it building with libc++ again.